### PR TITLE
[rename] Find ObjC class textual occurrences in string literals

### PR DESCRIPTION
**Pull request by hyp**
Occurrences match only when the string's text is exactly the same as class name.

rdar://33861206

### CODE REVIEW

File: lib/Tooling/Refactor/RenameIndexedFile.cpp [hunk 1]
```diff
@@ -333,12 +372,18 @@ static void findMatchingTextualOccurrences(
...
         CommentMatches.clear();
       }
-    } else if (!SelectorParsers.empty()) {
+    } else if (ScanNonCommentTokens) {
       for (auto &Parser : SelectorParsers) {
         if (Parser.handleToken(RawTok))
           MatchHandler(OldSymbolOccurrence::MatchingSelector,
                        Parser.SelectorLocations, Parser.SymbolIndex);
       }
+      for (auto &Parser : StringParsers) {
```
Comment (benlangmuir) [inline lib/Tooling/Refactor/RenameIndexedFile.cpp:381]: Why is StringParser `checkToken` while the selector parser uses `handleToken`?

**Review by benlangmuir — COMMENTED** [review]
What's your reasoning for only doing this for ObjC class names?  I'm okay with starting with the narrow case, but I'm curious what the motivation was there.

**Comment by hyp** [comment]
I think that class names are used in this kind of pattern more often than other declarations. Perhaps this happens with selectors too, but in general `@selector` is more common. Expanding this to other declarations would probably create too much noise, don't you think?

File: lib/Tooling/Refactor/RenameIndexedFile.cpp (hunk 1 above)
Comment (hyp) [reply lib/Tooling/Refactor/RenameIndexedFile.cpp:381]: Fixed

**Comment by benlangmuir** [comment]
After some thought, I think this is the right starting place.  We could fairly reliably search for selector names in strings except for the case of zero-argument selectors, but it would be surprising to have them behave differently than other selectors.  If we find a strong need in the future, we can think about extending this then.

**Review by benlangmuir — APPROVED** [review]

**Pull request merged**